src/server/controllers/match.controller.js

```diff
@@ -99,22 +100,28 @@ export function getMentorList(req, res, next) {
     })
     .then((careerFilteredList) => {
       return new Promise((resolve, reject) => {
-        let filteredList = {
-          full: [],
-          idList: [],
-        };
+        if (req.body.expertise === undefined) {
+          resolve(careerFilteredList.full);
+        } else if (req.body.expertise.length === 0) {
+          resolve(careerFilteredList.full);
```

Comment:
`
if (req.body.expertise === undefined || req.body.expertise.length === 0)
`
I guess since javascript has short circuit evaluation, if the left side of condition is true(which means expertise is undefined), the right side will not executed at all.